### ISSUE TITLE
InMemeoryAdapter Update object

### ISSUE THREAD

**Issue by DaveSussman** [ThatRendle/Simple.Data#122]
Nice work on the InMemoryAdapter; much nicer that the XmlmockAdapter, which I'll start to replace.

But, I think I've found a bug in the Update method, which you don't have a test case for. This is updating by passing in an object, as opposed to by named parameters, where you have:

```
    public override int Update(string tableName, IDictionary<string, object> data)
    {
        if (!_keyColumns.ContainsKey(tableName)) throw new InvalidOperationException("No key column(s) specified.");
        IDictionary<string, object> row = null;
        if (_keyColumns[tableName].Length == 1)
        {
            row =
                GetTable(tableName).Single(
                    d => d[_keyColumns[tableName][0]] == data[_keyColumns[tableName][0]]);
        }
        else
        {
            IEnumerable<IDictionary<string, object>> rows = GetTable(tableName);
            row = _keyColumns[tableName]
                .Aggregate(rows, (current, keyColumn) => current.Where(d => d[keyColumn] == data[keyColumn]))
                .Single();
        }
        UpdateRecord(data, row);
        return 1;
    }
```

Your equality comparison in searching for the correct row is failing because these are objects. This should be using .Equals, eg:

```
            row =
                GetTable(tableName).Single(
                    d => d[_keyColumns[tableName][0]].Equals(data[_keyColumns[tableName][0]]));
```

Or am I missing something obvious? I presume it's similar for multiple keys, although I've only tested the single key scenario, using:

```
public class User
{
    public int UserId { get; set; }
    public string UserName{ get; set; }
}

    [Test]
    public void should_update_object_with_single_key()
    {
        var adapter = new InMemoryAdapter();
        adapter.SetKeyColumn("Users", "UserId");

        Database.UseMockAdapter(adapter);
        dynamic db = Database.Open();

        db.Users.Insert(UserId: 1, UserName: "Marcus");
        db.Users.Insert(UserId: 2, UserName: "Per");

        User user = (User)db.Users.FindByUserId(1);
        user.UserName = "Markus";

        db.Users.Update(user);

        User updateduser = (User)db.Users.FindByUserId(1);

        Assert.AreEqual(user.UserName, updateduser.UserName);
    }
```

At the moment I'm a Git newbie and didn't think this was the right time to try and learn pull/push/patch/etc. Oh, and it's not critical; this part of the app is nowhere near done yet, so a failing test is no biggie.

**Comment by ThatRendle** [comment]
I'll add a test for this and get it passing. Cheers!

**Comment by ThatRendle** [comment]
Yep, it was the == instead of Equals that caused that. Fixed, on NuGet now, 0.11.2.
